src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -395,7 +395,8 @@ private static bool IsValidExtensionReceiverParameter(ParameterSymbol thisParam)
                 return false;
             }
 
-            if (thisParam.RefKind is RefKind.In or RefKind.RefReadOnlyParameter && thisParam.Type.TypeKind != TypeKind.Struct)
+            if (thisParam.RefKind is RefKind.In or RefKind.RefReadOnlyParameter
+                && (!thisParam.Type.IsValueType || thisParam.TypeWithAnnotations.TypeKind == TypeKind.TypeParameter))
```

Comment:
>TypeWithAnnotations [](http://example.com/codeflow?start=61&length=19)

Is there a particular reason to use ```TypeWithAnnotations``` here and ```Type``` in the previous condition? Also, consider using a pattern to reuse value returned by ```Type```:
```
thisParam.Type is not { ValueType: true, TypeKind: not TypeKind.TypeParameter}
```